transform.go

```diff
@@ -99,7 +99,7 @@ func InjectOwner(owner Owner) Transformer {
 	}
 }
 
-func isClusterScoped(kind string) bool {
+func IsClusterScoped(kind string) bool {
```

Comment:
As you can tell from the TODO, I always considered this function a kind of placeholder, until I (or someone, hint, hint!) could discover The Right Way. I'm happy to export it, but I wish it wasn't so "hard-codey". :smile:

Reply:
Brought this up to #controller-runtime Slack channel. [link](https://kubernetes.slack.com/archives/C02MRBMN00Z/p1674115565750249)

I guess we can simply use the Discovery API (https://pkg.go.dev/k8s.io/client-go/discovery) to query the APIResource by GVK. 

However, there's a performance aspect to think about: every call to `isClusterScoped` shouldn't go and actually hit the API server. I asked in that Slack thread about a cache+informer+watcher mechanism for APIResources.

Reply:
@jcrossley3 https://github.com/kubernetes-sigs/controller-runtime/pull/2136

That is a new PR that exposes a functionality `IsAPINamespaced`. Not sure if that can be used as-is, since it requires a scheme and a restMapper.

Reply:
Anyway, I think we can close this PR to make `isClusterScoped` exposed. I will use the `IsApiNamespaced` from https://github.com/kubernetes-sigs/controller-runtime/pull/2136 in my usecase.